core/controllers/admin_test.py

```diff
@@ -1109,18 +1109,69 @@ def test_send_dummy_mail(self):
 
         with self.swap(feconf, 'CAN_SEND_EMAILS', True):
             generated_response = self.post_json(
-                '/sendDummyMailToAdminHandler', payload={},
+                '/senddummymailtoadmin', payload={},
                 csrf_token=csrf_token, expected_status_int=200)
             self.assertEqual(generated_response, {})
 
         with self.swap(feconf, 'CAN_SEND_EMAILS', False):
             generated_response = self.post_json(
-                '/sendDummyMailToAdminHandler', payload={},
+                '/senddummymailtoadmin', payload={},
                 csrf_token=csrf_token, expected_status_int=400)
             self.assertEqual(
                 generated_response['error'], 'This app cannot send emails.')
 
 
+class UpdateUsernameHandlerTest(test_utils.GenericTestBase):
+    """Tests for updating usernames."""
+
+    def setUp(self):
+        super(UpdateUsernameHandlerTest, self).setUp()
+        self.signup(self.ADMIN_EMAIL, self.ADMIN_USERNAME)
+
+    def test_update_username(self):
+        self.login(self.ADMIN_EMAIL, is_super_admin=True)
+        csrf_token = self.get_new_csrf_token()
+
+        current_username, new_username = self.ADMIN_USERNAME, 'newUsername'
+        non_existent_username, long_username = 'invalid', 'a' * 31
+        user_id = self.get_user_id_from_email(self.ADMIN_EMAIL)
+
+        # The new username must not be longer than 30 characters.
+        self.put_json(
+            '/updateusername',
+            payload={
+                'current_username': current_username,
+                'new_username': long_username},
+            csrf_token=csrf_token,
+            expected_status_int=400)
```

Comment:
If you just check for 400, the tests are weak and don't differentiate observably between the different cases. Please check the error messages too.

Reply:
Done.